aws_embedded_metrics/logger/metrics_context.py

```diff
@@ -162,6 +172,14 @@ def create_copy_with_context(self) -> "MetricsContext":
             self.namespace, new_properties, new_dimensions, new_default_dimensions
         )
 
+    def create_copy_with_context_with_dimensions(self) -> "MetricsContext":
```

Comment:
In the Java lib you used `createCopyWithContextWithoutDimensions()`. I think you should match the implementation of these methods as much as possible across libraries.

Reply:
Resolved. Would keep the code since different libs have different default bahaviors.